caddyfile.go

```diff
@@ -66,7 +66,11 @@ func parseApp(d *caddyfile.Dispenser, _ interface{}) (interface{}, error) {
 				if app.Domains == nil {
 					app.Domains = make(map[string][]string)
 				}
-				app.Domains[zone] = names
+				if app.Domains[zone] != nil {
+					app.Domains[zone] = append(app.Domains[zone], names...)
+				} else {
+					app.Domains[zone] = names
+				}
```

Comment:
```suggestion
					app.Domains[zone] = append(app.Domains[zone], names...)
```

Does this work? I think you can append to a nil slice and it will just work...

Reply:
That definitely seems to work. Strange, [the documentation for `append`](https://pkg.go.dev/builtin#append) makes no mention of that. 

Anyway, that makes it even simpler!